### ISSUE TITLE
Learn mode directs player to the wrong actions for certain seeds

### ISSUE THREAD

**Issue by westonb7** [reputage/seedQuest#359]
**Description**
Learn mode directs the player to perform the wrong actions for some seeds. This was brought up by Rouven in the SeedQuest slack channel. His example was the seed "ABCD**E**E07", learn mode directs the player to perform the actions to reproduce the seed "ABCD**F**E07" instead.
This can also be seen with the seed "**E**EFF**E**EFF", learn mode will direct the player to perform the actions to reproduce the seed "**F**EFF**F**EFF" instead of the intended seed.

Both seeds "ABCD**E**E07" and "**E**EFF**E**EFF" can be recovered successfully in recover mode, if the right actions are performed. (The correct actions for "**E**EFF**E**EFF" are: **stew** -> polish -> polish -> **stew** -> polish -> polish) But in Learn mode, even if the right actions are performed, the wrong seed will be recovered regardless.

**To Reproduce**
1. Enter a custom seed (It's easy to reproduce and see this with "**E**EFF**E**EFF"
2. Play learn mode with the custom seed.
3. Watch the recovered seed as it's reproduced by player actions.

**Expected Behavior**
With the seed "**E**EFF**E**EFF" in the islands, the player should be prompted to perform the actions:
Cauldron: **Cook stew** -> Cauldron: Polish -> Cauldron: Polish -> Cauldron: **Cook stew** -> Cauldron: Polish -> Cauldron: Polish 

**Actual Behavior**
The player is instead prompted to perform the actions:
Cauldron: **Polish** -> Cauldron: Polish -> Cauldron: Polish -> Cauldron: **Polish** -> Cauldron: Polish -> Cauldron: Polish 

(or "polish cauldron" six times in a row)

**Screenshots**
<img width="713" alt="Screen Shot 2019-04-03 at 6 02 31 PM" src="https://user-images.githubusercontent.com/30869523/55520787-0e086080-563b-11e9-85d2-0abe2ed62cc1.png">
Here in 'Learn' mode, even if I perform the 'cook stew' action, it will still act as if I had recovered a seed using the 'polish' action.

<img width="711" alt="Screen Shot 2019-04-03 at 6 02 39 PM" src="https://user-images.githubusercontent.com/30869523/55520795-152f6e80-563b-11e9-824a-e76079c64b3b.png">
Here I've recovered the seed "FEFFFEFF" in learn mode, even though the custom seed I entered was "EEFFEEFF".

<img width="711" alt="Screen Shot 2019-04-03 at 6 04 21 PM" src="https://user-images.githubusercontent.com/30869523/55520798-182a5f00-563b-11e9-876a-96d765f2be88.png">
Here in 'recover' mode, if I perform the 'cook stew' action, it will recover the seed I actually want.

<img width="711" alt="Screen Shot 2019-04-03 at 6 04 29 PM" src="https://user-images.githubusercontent.com/30869523/55520800-1a8cb900-563b-11e9-9748-b5e24b9fe8c5.png">
Here in 'recover' mode, I can successfully recover "EEFFEEFF" with the correct actions.

**Device Info (please complete the following information):**
 - OS: Sierra
 - Version v0.1.1
 - Device: Mac

**Additional context**
I'm not sure if the problem is in the front end or the back end, but I added an extra test to SeedToByte tests in a new branch ('rouven-bug-testing') to specifically test for converting the above seeds into actions and back into seeds, which worked successfully.

**Comment by westonb7** [comment]
Update: it seems this bug has to do with seeds that result in a path that causes the player to need to interact with the same interactable multiple times in a row. For example, in Islands, the seed "ce7dce7d" should result in interacting with the cauldron interactable three times with three different actions, however in learn mode, the player is directed to perform the same action on the cauldron three times, and results in recovering the incorrect seed "de7bde7b" instead. 

In other words, the problem is that learn mode will not account for different actions being performed on the same interactable multiple times in a row, for some reason.

I'm still not sure what the exact source of the problem is, but may be related to the 'GoToNextInteractable()' function in the InteractablePath script.

**Comment by michaelmendoza** [comment]
Alright, good insight. I'll investigate further.

**Comment by westonb7** [comment]
I did some work on this issue recently, and it turns out that it was actually several different issues across multiple scripts that were all contributing to the same problem. 

- One bug caused the interactable path to think that if an interactable is being used multiple times, it would use the same action over and over again.

- Another bug caused Learn mode to log the actions stored in the interactable path, and not actually log the actions being performed by the player while in Learn mode.

- Another set of bugs caused an issue with the interactable preview to display the wrong prefab and wrong action name when the same interactable was in the interactable path more than once

I've made some changes that should fix some of these problems, but it's possible that the fixes may have created more unseen issues.
